app/components/articleSearch/reducer.tsx

```diff
@@ -45,15 +46,30 @@ export function reducer(
     case ACTION_TYPES.ARTICLE_SEARCH_SUCCEEDED_TO_GET_PAPERS: {
       const payload: SearchResult = action.payload;
 
+      const sortedFosList = payload.data.aggregation.fosList.sort((a, b) => {
+        if (state.fosFilter.includes(a.id) && !state.fosFilter.includes(b.id)) {
```

Comment:
sort function is just the same with below. 
you can make it one unified function.